### PR TITLE
Initial implementation of HTTPServer

### PR DESCRIPTION
**Pull request by pandabear15**
An initial implementation of HTTPServer via Django. Currently does not obey the API contract and simply indicates that the server can respond to requests.

A future PR will need to modify the backend CI to run python manage.py test in the correct directory.

### CODE REVIEW

File: backend/src/httpserver/httpserver/settings.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "django-insecure-+twskqgt5oq5#g7ar1w4jac7ajfmamsi^xx#7nl%svqadp)262"
```
Comment (jspaniac) [inline backend/src/httpserver/httpserver/settings.py:23]: Should this be posted to the repository directly?

File: backend/src/httpserver/httpserver/settings.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    },
+]
+
+WSGI_APPLICATION = "httpserver.wsgi.application"
+
+
+# Database
+# https://docs.djangoproject.com/en/4.1/ref/settings/#databases
+
+DATABASES = {
```
Comment (jspaniac) [inline backend/src/httpserver/httpserver/settings.py:77]: Are we planning on using our Bitio databse or this sqlite one?

File: backend/src/httpserver/httpserver/urls.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+Examples:
+Function views
+    1. Add an import:  from my_app import views
+    2. Add a URL to urlpatterns:  path('', views.home, name='home')
+Class-based views
+    1. Add an import:  from other_app.views import Home
+    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+Including another URLconf
+    1. Import the include() function: from django.urls import include, path
+    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
```
Comment (jspaniac) [inline backend/src/httpserver/httpserver/urls.py:14]: Probably will want to delete this boilerplate comment eventually? Could also keep it around for posterity if we need to add one of these things

**Review by jspaniac — APPROVED** [review]
I'll approve just so we can get something up and running for Tues, but we should probably take a closer look at the comments left

File: backend/src/httpserver/httpserver/settings.py (hunk 2 above)
Comment (pandabear15) [reply backend/src/httpserver/httpserver/settings.py:77]: This is just the database that comes with Django and we are not obligated to use it

File: backend/src/httpserver/httpserver/urls.py (hunk 3 above)
Comment (pandabear15) [reply backend/src/httpserver/httpserver/urls.py:14]: Given I'm still not super comfortable with django it's good for me at least :P

File: backend/src/httpserver/httpserver/settings.py (hunk 1 above)
Comment (pandabear15) [reply backend/src/httpserver/httpserver/settings.py:23]: Based on how I'm handling hosting this should not be a problem

**Pull request merged**